src/appleseed-max-impl/appleseedrenderer/appleseedrenderer.cpp

```diff
@@ -207,10 +207,6 @@ void AppleseedRendererPBlockAccessor::Get(
         v.i = static_cast<int>(settings.m_output_mode);
         break;
 
-      case ParamIdProjectPath:
```

Comment:
What are the consequences on functionality and backward compatibility of removing these two `case` blocks?

Reply:
This question still holds :)

Reply:
Sorry I missed this one. Let me check, it didn't change the functionality, but I didn't try backward compatibility test.

Reply:
@dictoon, I checked and it works fine. It opens the file from master version without any problem, settings are all there and everything.
It's still not clear why that section started to crash in 2018-19, but it looks like a fix, I don't know.

Reply:
I just don't understand how the project path is saved if `ParamIdProjectPath` is no longer handled. It looks like the project path should no longer be written into project files?

Reply:
Hi @dictoon 
Setting project path variable happens in `AppleseedRendererPBlockAccessor::Set` function, and it has not been changed. So saving settings should be ok.
I thought loading may brake but it's ok too, so I think this change to Get methods seems harmless.

Reply:
From what I understand, `AppleseedRendererPBlockAccessor::Get()` extracts values from `owner` (of type `ReferenceMaker*`) and stores them in `v` (of type `PB2Value&`), so it seems that this corresponds to _saving_ settings. Conversely, `Set()` seems to correspond to _loading_ settings.

So as far as I can tell your change here should impact _saving_ project files, not loading them.

I still don't understand why everything would continue working with these changes (i.e. with hadling of `ParamIdProjectPath` removed). If it's indeed the case that things continue to work 100%, then isn't there a lot of other redundant code that could also be removed?

What's the logic?

Reply:
This is how I understand it:
Saving and loading of renderer parameters is still happening in the `RendererSettings::save/load` functions, which saves, let's say `m_project_file_path` variable. Value of that variable is set in the `PBlockAccessor::Set()` function. When I enter the path into UI and hit enter `Set()` function is called and sets the value from `v.s` to `m_project_file_path`. After that our old saving part should correctly save the path.

Other thing is that these new pblock properties that I added recently are also saving/loading by themself. But I added `P_TRANSIENT` keyword to all of them, which according to docs stops them from being stored. Here's the quote "do not store actual value, PBAccessor-derived."

Also I added `P_CALLSETS_ON_LOAD` flag to the pblock header, which after loading the file should call `Set()` function to all the pblock properties.

I agree that it's not clear why everything works after removing two property getters. My guess is that properties are still saved on disk and get loaded, and that happens before `Get()/Set()` functions are called, so they only update the values with what our Save/Load functions get and put into renderersettings class.

Reply:
About redundent code - we could probably remove all our `RendererSettings::Save/Load` functions and access all the properties directly from Renderer object but that would break backward compatibility.

Reply:
Thanks for the detailed explanation, much appreciated.

What I now understand better is that this code has nothing to do with loading/saving settings, instead it appears to be about feeding the paramblocks behind the user interface widgets.

What I still don't understand is why we can safely remove the code that handles project file paths but not code related to other settings.

Let's merge this. I'd suggest to keep investigating whether more code can be removed or not.

Reply:
Crap, I forgot to merge this PR. Wandering what happens with Mango3's pr if I merge this now...